test/common/index.js

```diff
@@ -271,9 +271,17 @@ Object.defineProperty(exports, 'hasFipsCrypto', {
 });
 
 {
-  const localRelative = path.relative(process.cwd(), `${exports.tmpDir}/`);
-  const pipePrefix = exports.isWindows ? '\\\\.\\pipe\\' : localRelative;
-  const pipeName = `node-test.${process.pid}.sock`;
+  let pipePrefix;
+  if (exports.isWindows) {
+    pipePrefix = '\\\\.\\pipe\\';
+  } else {
+    const localRelative = path.relative(process.cwd(), `${exports.tmpDir}`);
+    pipePrefix = localRelative + path.sep;
+  }
+
+  const randHex = Math.random().toString(16).slice(2);
```

Comment:
`Math.random()` in parallel tests introduces a chance of collisions and hard to reproduce flakes.

Reply:
@richardlau I *think* if there are flakes the path name should be reported and we can reproduce that by hard-coding that name here...

Not sure which one would introduce less collisions though, `Math.random()` or pid + filename initials? Or just mix them all together?

Reply:
Since we run a few thousands of tests in each run, and the collision probability of `Math.random` is ~2<sup>30</sup> I think we're safe.
Also IMHO less code => less hard to grok failures.

P.S. `pid` is an arbitrary number and it did lead to hard to reproduce flakes, but we managed to come up with a theory.

tl;dr let's try to leave clues:
1) a comment
2) I'll try to think of a way to output a clue in case of EADDRINUSE errors